modules/Sources/Dependencies/FiatPriceClient/FiatPriceClient.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  FiatPriceClient.swift
+//
+//
+//  Created by Matthew Watt on 10/1/23.
+//
+
+import ComposableArchitecture
+import Foundation
+import Models
+import Utils
+
+public struct FiatPriceClient {
+    public var getZcashPrice: (NighthawkSetting.FiatCurrency) async throws -> Double?
+}
+
+extension FiatPriceClient: DependencyKey {
+    struct FiatPriceApiResponse: Codable, Equatable {
+        let data: [String: Double]
+        
+        enum CodingKeys: String, CodingKey {
+            case data = "zcash"
+        }
+    }
+    
+    public static let liveValue = Self(
+        getZcashPrice: { currency in
+            guard currency != .off else { return nil }
+                
+            let simplePriceUrl = URL.coinGeckoApi
+                .appending(path: "simple/price")
+                .appending(
+                    queryItems: [
+                        .init(name: "ids", value: "zcash"),
+                        .init(name: "vs_currencies", value: currency.rawValue)
+                    ]
+                )
+            
+            let (data, _) = try await URLSession.shared.data(from: simplePriceUrl)
```

Comment:
[Blocking] I have a concern about the timing of these calls. If the app polls the value in a predictable way, this fiat conversion is an excellent vector for timing attacks from an adversary. The pace of a service like this can be used to  profile a user an identify it as a NH user and use this also as a way of knowing when a target has started using the app.

This has a sort of simple solution that is randomizing how this value is refreshed over a range of time, so that it can't be statistically determined.

Reply:
By default, fiat currency API call is disabled in Nighthawk, aligning with our security-first approach to third-party services. Users may decide to opt-in to this feature, ensuring protection against possible, predictable timing-based attacks.

When a Nighthawk user chooses to enable fiat currency display, the price API is only polled at the start of the app session, typically lasting 5-15 minutes. The term 'using' the app refers to users broadcasting a transaction, not just opening the app. Users may open the app for various purposes, such as sharing their receiving address or viewing transaction history, which doesn't convert to user-tracking due to the open nature of the pricing API used by wallets in the cryptocurrency ecosystem.

This approach distinguishes Nighthawk from other wallets that constantly poll real-time data during public interactions with the blockchain, reducing the risk of user tracking. However, randomizing refreshed values will result in showing inaccurate or outdated fiat currency rates to the user. This is a tradeoff for users that decide to opt-in to viewing ZEC price within the app.

Users are also notified of the following text before they decide to opt-in to this feature:

"Choose your local currency so that we can show you a close estimate of how much your ZEC is worth.

When you first open the app, the exchange rate is fetched from CoinGecko API v3 and used to calculate values. Your fund amounts are never revealed to any server."

NOTE: Please report any possible vulnerabilities or attack vectors following our disclosure policy in the project's README.

Reply:
I consider this comment resolves my comment. Thanks @nighthawk24